### PR TITLE
assert before saving on disk to avoid deleting the file

### PR DESCRIPTION
**Pull request by Isalia20**
Fixes the issue in #6294  by asserting that no tensor should have disk device open before saving. Otherwise it causes the file to delete as described in the issue. Including the code to reproduce from the issue for convenience:
```
from tinygrad import Tensor
from tinygrad.nn.state import safe_save, safe_load, get_state_dict
import os
os.environ['GPU'] = '1'

t1 = Tensor([1, 2, 3, 4, 5])
state_dict = get_state_dict(t1)
safe_save(state_dict, "model.safetensors")
state_dict2 = safe_load("model.safetensors")
safe_save(state_dict2, "model.safetensors")   # Deletes the existing file and does not replace it!
state_dict3 = safe_load("model.safetensors")   # Error: No such file
```

### CODE REVIEW

**Comment by Isalia20** [comment]
Get an error on the linter but I think it's a false one:
`tinygrad/nn/state.py:55: error: "Allocator" has no attribute "device"  [attr-defined]`

Should I add type ignore?

**Comment by geohot** [comment]
No. That's real.

**Comment by Isalia20** [comment]
I've changed it to device.count directly. This should work

**Comment by Isalia20** [comment]
should be ready now

File: tinygrad/device.py [hunk 1]
```diff
@@ -184,8 +184,9 @@ def compile_cached(self, src:str) -> bytes:
     return lib
 
 class Compiled:
-  def __init__(self, device:str, allocator:Allocator, renderer:Optional[Renderer], compiler:Optional[Compiler], runtime, graph=None):
-    self.dname, self.allocator, self.compiler, self.runtime, self.graph = device, allocator, compiler or Compiler(), runtime, graph
+  def __init__(self, device:str, allocator:Allocator, renderer:Optional[Renderer],
+               compiler:Optional[Compiler], runtime, graph=None, count=0):
+    self.dname, self.allocator, self.compiler, self.runtime, self.graph, self.count = device, allocator, compiler or Compiler(), runtime, graph, count
```
Comment (geohot) [inline tinygrad/device.py:189]: How is count not always 0?

Comment (Isalia20) [reply tinygrad/device.py:189]: I added it so mypy wouldn't give the error on Device not having the count, but I agree it's not the cleanest of solutions. I'll close this PR now and come up with something better

**Pull request closed without merging**